src/main/webapp/detailsScript.js

```diff
@@ -0,0 +1,104 @@
+let API_KEY;
+// Obtain the API_KEY from local file.
+fetch('apiKey.json')
+  .then(response => response.json())
+  .then(jsonResponse => {
+    API_KEY = jsonResponse.apiKey;
+    getBookDetails();
+  });
+
+// To get results from UK publishing houses.
+const country = 'UK';
+
+// Obtain the book ID from the query string. This is the Google Books API ID.
+let id = parseQueryString(window.location.search).id;
+
+/**
+ * Function that is automatically triggered when the user is redirected to the bookDetails page.
+ * Extracts the Google Books API ID of the individual book from the URL and performs a GET request
+ * with that ID to obtain a JSON with all the details about that specific book.
+ */
+
+function getBookDetails() {
+  
+  let URL = `https://www.googleapis.com/books/v1/volumes/${id}`;
+  URL += `?key=${API_KEY}`;
+  URL += `&country=${country}`;
+  $.ajax({
+    url: URL,
+    type: 'GET',
+    success: function(result) {
+      console.log(result);
+      displayDetails(result);
+    },
+    error: function(error) {
+      alert(`Error ${error}. Please try again.`);
+    },
+  });
+}
+
+/**
+ * Completes all the fields on the page with details from the JSON answer of the API.
+ * @param {JSON} bookJSON 
+ */
+
+function displayDetails(bookJSON) {
+  const volumeInfo = bookJSON.volumeInfo;
+  document.getElementById('title-header').innerText = volumeInfo.title;
+  document.getElementById('authors').innerText = 
+    `Authors: ${volumeInfo.authors}`;
+  document.getElementById('published-date').innerText = 
+    `Published date: ${volumeInfo.publishedDate}`;
+  document.getElementById('category').innerText =
+    `Categories: ${volumeInfo.categories}`;
+  document.getElementById('publisher').innerText =
+    `Publishing house: ${volumeInfo.publisher}`;
+  document.getElementById('page-count').innerText =
+    `Page count: ${volumeInfo.pageCount}`;
+  document.getElementById('description').innerText = volumeInfo.description;
+  document.getElementById('identifiers').innerText = getStringIdentifiers(volumeInfo.industryIdentifiers);
+}
+
+/**
+ * identifiers array contains objects like:
+ * {
+ *      "type": "ISBN_10",
+ *      "identifier": "0415286514"
+ * }
+ * The function transform them into a string.    
+ * @param {Array of JSONs} identifiers 
+ */
+
+function getStringIdentifiers(identifiers) {
```

Comment:
nit: You could use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map and https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join to write this in a more functional style, rather than appending to a string.

Reply:
Done